compiler/noirc_frontend/src/hir/def_collector/dc_crate.rs

```diff
@@ -476,26 +471,34 @@ impl DefCollector {
             );
         }
 
-        if error_on_usage_tracker {
-            Self::check_usage_tracker(context, crate_id, &mut errors);
+        if error_on_unused_items {
+            Self::check_unused_items(context, crate_id, &mut errors);
         }
 
         errors
     }
 
-    fn check_usage_tracker(
-        context: &Context,
+    fn check_unused_items(
+        context: &mut Context,
         crate_id: CrateId,
         errors: &mut Vec<(CompilationError, FileId)>,
     ) {
-        let unused_imports = context.def_interner.usage_tracker.unused_imports().iter();
+        // Assume `main` is used so we don't warn on that
+        let root = ModuleId { krate: crate_id, local_id: context.def_maps[&crate_id].root };
+        let main = Ident::new("main".to_string(), Span::default());
+        context.def_interner.usage_tracker.mark_as_used(root, &main);
```

Comment:
Have you tested this with e.g. contract crates to ensure the entry functions there don't warn?

Reply:
Oh, I didn't know abot that. What are contract entry functions?

Reply:
Ah, I should have searched a bit more, I guess they are `fn entrypoint` 😅

Reply:
I just checked and even though they are private, they don't error. They are annotated with `aztec`, and looking a bit into the macro code it seems they are changed to `pub fn` 😮‍💨

Reply:
Can you assure this is still the case for contract functions not associated with `aztec`?
IIRC entry point functions in contracts need to be annotated with `#[entry_point]` or something similar:
```rs
contract foo {
    #[entry_point]
    fn bar() {}
}
```

Reply:
Okay, it seems all contract methods are public entrypoints, so for those we shouldn't produce an unused warning. I'll push that change.

Reply:
Done! Also made sure that case is covered in a test. I also checked again in Aztec-Packages and the functions that were giving a warning no longer give one.